From 1c7bf964655ecea3e0d33db4d050345c9411ab3a Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Tue, 28 Feb 2023 11:22:22 +0100
Subject: [PATCH] arm64: dts: st: enable management of the Cortex-M33 on
 stm32mp257f-ev1

Enable the IPCC and the remoteproc for the management of the cCortex-M33.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I87c3e1c17f07e69631f92956f6a477b83ebeb825
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index 69a9c5f0b868..5d5d7c652455 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -69,6 +69,10 @@ memory@80000000 {
 	};
 };
 
+&a35ss_syscfg {
+	status = "okay";
+};
+
 &arm_wdt {
 	timeout-sec = <32>;
 	status = "okay";
@@ -167,6 +171,20 @@ &i2c2 {
 	/delete-property/dma-names;
 };
 
+&ipcc1 {
+	status = "okay";
+};
+
+&m33_rproc {
+	mboxes = <&ipcc1 0x100>, <&ipcc1 0x101>, <&ipcc1 2>;
+	mbox-names = "vq0", "vq1", "shutdown";
+	memory-region = <&cm33_cube_fw>, <&cm33_cube_data>,
+			<&ipc_shmem_1>, <&vdev0vring0>,
+			<&vdev0vring1>, <&vdev0buffer>;
+	st,syscfg-nsvtor = <&a35ss_syscfg 0xa8 0xffffff80>;
+	status = "okay";
+};
+
 &ommanager {
 	memory-region = <&mm_ospi1>;
 	memory-region-names = "mm_ospi1";
-- 
2.39.5

